src/query/procedure/module.cpp

```diff
@@ -83,6 +84,19 @@ void BuiltinModule::AddTransformation(std::string_view name, mgp_trans trans) {
 
 namespace {
 
+auto WithUpgradedLock(auto *lock, const auto &function) {
+  lock->unlock_shared();
+  try {
+    function();
+    // There's no finally in C++, but we have to return our original READ lock
```

Comment:
Why don't you use `OnScopeExit` that passes a lambda which calls `lock->lock_shared()`

Reply:
that absolutely works, nice catch!